### PR TITLE
WIP: Document that children are not stopped on parent restart, #25556

### PR DESCRIPTION
**Pull request by patriknw**
Refs #25556

@svroonland please review

### CODE REVIEW

File: akka-docs/src/main/paradox/typed/fault-tolerance.md [hunk 1]
```diff
@@ -74,6 +83,39 @@ Java
 
 Each returned behavior will be re-wrapped automatically with the supervisor.
 
+## Child actors remain when restarting
+
+Child actors are not influenced when the parent actor is restarted. The restarted parent instance will have the same
+children as before the failure.
```
Comment (ktoso) [inline akka-docs/src/main/paradox/typed/fault-tolerance.md]: 👍 as discussed in a call, this is good, and less confusing than the state of untyped, indeed good to only document it.

**Review by ktoso — APPROVED** [review]
LGTM :)

File: akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala [hunk 2]
```diff
@@ -54,4 +55,51 @@ object SupervisionCompileOnly {
...
+  //#restart-stop-children
+
+  //#restart-keep-children
+  def parent2: Behavior[String] = {
+    Behaviors.setup { ctx ⇒
+      val child1 = ctx.spawn(child(0), "child1")
+      val child2 = ctx.spawn(child(0), "child2")
+
+      // supervision strategy inside the setup to not recreate children on restart
+      Behaviors.supervise {
```
Comment (svroonland) [inline akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala]: Nice

**Review by svroonland — APPROVED** [review]
Looks good to me, nicely documented how to avoid recreating children on restart by moving the supervision.

File: akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala [hunk 3]
```diff
@@ -54,4 +55,51 @@ object SupervisionCompileOnly {
...
+        Behaviors.receiveMessage[String] { msg ⇒
+          // there might be bugs here...
+          val parts = msg.split(" ")
+          child1 ! parts(0)
+          child2 ! parts(1)
+          Behaviors.same
+        }.receiveSignal {
+          // stop children when restarted, new instances are created from setup
+          case (c, _: PreRestart) ⇒
+            c.children.foreach(c.stop)
```
Comment (patriknw) [inline akka-actor-typed-tests/src/test/scala/docs/akka/typed/supervision/SupervisionCompileOnly.scala]: This approach doesn't work as noted in https://github.com/akka/akka/issues/25556#issuecomment-420765175
The name isn't free until receiving `Terminated`, hm that becomes complicated.

**Comment by patriknw** [comment]
We have a way forward here, as discussed in ticket #25556. Closing this for now since it requires more than documentation. Some of these examples might be incorporated in later PR.

**Pull request closed without merging**